tabsforeditorswidget.cpp

```diff
@@ -162,27 +219,75 @@ void TabsForEditorsWidget::prevTabAction()
 {
   int currentIndex = m_tabWidget->currentIndex();
   if (currentIndex >= 1)
-  {
     m_tabWidget->setCurrentIndex(currentIndex - 1);
-  }
   else
-  {
     m_tabWidget->setCurrentIndex(m_tabWidget->count() - 1);
-  }
 }
 
 void TabsForEditorsWidget::nextTabAction()
 {
   int currentIndex = m_tabWidget->currentIndex();
   if (currentIndex < m_tabWidget->count() - 1)
-  {
     m_tabWidget->setCurrentIndex(currentIndex + 1);
-  }
   else
-  {
     m_tabWidget->setCurrentIndex(0);
-  }
 }
+
+void TabsForEditorsWidget::handleTabRightButtonClick(int tabIndex, QPoint &position)
+{
+    m_currentTabIndex = tabIndex;
+
+    m_contextMenu->exec(mapToGlobal(position));
+}
+
+void TabsForEditorsWidget::handleTabMiddleButtonClick(int tabIndex, QPoint &)
+{
+    handleTabCloseRequested(tabIndex);
+}
+
+void TabsForEditorsWidget::setupContextMenu()
+{
+    m_contextMenu = new QMenu(this);
+
+    m_closeTabAction = new QAction(tr("Close Tab"), this);
+    m_closeAllTabsExceptThisAction = new QAction(tr("Close All BUT This Tab"), this);
+    m_closeTabsToRightAction = new QAction(tr("Close Tabs to the Right"), this);
+    m_closeAllTabsAction = new QAction(tr("Close All Tabs"), this);
+
+    m_signalMapper = new QSignalMapper(this);

```

Comment:
The reason for that is because the `triggered()` signal of `QMenu` has either no parameters, or a `QAction*` parameter. 

AFAIK It would be possible, however, to add the index to the respective `QAction` object via `QAction::setData(...)` and change each of the action handling slots' signature.